bundles/io/org.eclipse.smarthome.io.transport.mqtt/src/main/java/org/eclipse/smarthome/io/transport/mqtt/MqttService.java

```diff
@@ -217,212 +209,57 @@ public boolean addBrokerConnection(MqttBrokerConnection connection) {
      * Additional properties are "user","pwd","qos","retain","lwt","keepAlive","clientId", please read the
      * service configuration documentation for a detailed description.
      *
-     * @param brokerConnectionConfig The configuration key-value map.
+     * @param config The configuration key-value map.
      * @return Returns the created broker connection or null if there is already a connection with the same name.
-     * @throws ConfigurationException Most likely your provided name and url are invalid.
+     * @throws ConfigurationException Most likely your provided host is invalid.
      * @throws MqttException
      */
-    public MqttBrokerConnection addBrokerConnection(Map<String, String> brokerConnectionConfig)
-            throws ConfigurationException, MqttException {
-        // Extract mandatory fields
-        String brokerID = brokerConnectionConfig.get(NAME_PROPERTY);
-        if (StringUtils.isBlank(brokerID)) {
-            throw new ConfigurationException("MQTT Broker property 'name' is not provided");
-        }
-        brokerID = brokerID.toLowerCase();
+    public synchronized MqttBrokerConnection addBrokerConnection(@NonNull String brokerID,
```

Comment:
@kaikreuzer Do I remember correctly that you allowed `@Nullable` on return types to enable checks for the consumer code?
If it is allowed I would prefer to add it here.

Reply:
Hm, NonNull is the java standard. I agree that it sometimes makes sense to add it, as a way to document parameters, for instance Integer: A consumer might feel like he always needs to provide a value, even if `null` would be absolutely acceptable.

But for return types I would say Nullable is not reasonable.

Reply:
Please read https://github.com/eclipse/smarthome/pull/4080

Reply:
Nah a twenty page long discussion doesn't help :) A link or quote of the developer guidelines would convince me of course :D

Reply:
There isn't a developer guidelines about the usage of nullness annotation. At least we didn't fully agree.

Reply:
At least everything that has been agreed on, should be put in the developer guidelines. Should I open an issue?

Reply:
I don't think so. I have added a summary here:
https://github.com/eclipse/smarthome/wiki/Type-annotations